diff -Nur a/src/lib/dhcp/dhcpcd b/src/lib/dhcp/dhcpcd
--- a/src/lib/dhcp/dhcpcd	2016-06-21 16:33:14.845853066 +0200
+++ b/src/lib/dhcp/dhcpcd	2016-06-21 16:34:41.750284002 +0200
@@ -12,7 +12,7 @@
         return 1
     fi
     # If using own dns, tell dhcpcd to NOT replace resolv.conf
-    [[ $DNS ]] && options+=" -C resolv.conf"
+    [[ $DNS ]] && options+=" -C $STATE_DIR/resolv.conf"
     if ! do_debug do_readable dhcpcd -$1 -q -t "${TimeoutDHCP:-30}" $options "$Interface"; then
         report_error "DHCP IPv$1 lease attempt failed on interface '$Interface'"
         return 1
