diff -Nur a/services/netctl-ifplugd@.service b/services/netctl-ifplugd@.service
--- a/services/netctl-ifplugd@.service	2016-06-23 11:37:51.407834931 +0200
+++ b/services/netctl-ifplugd@.service	2016-06-23 11:38:16.051957135 +0200
@@ -5,7 +5,7 @@
 After=sys-subsystem-net-devices-%i.device network-pre.target
 
 [Service]
-ExecStart=/usr/bin/ifplugd -i %I -r /etc/ifplugd/netctl.action -bfIns
+ExecStart=/sbin/ifplugd -i %I -r /etc/ifplugd/netctl.action
 
 [Install]
 WantedBy=multi-user.target
