diff -Nur a/atspi/atspi-gmain.c b/atspi/atspi-gmain.c
--- a/atspi/atspi-gmain.c	2016-08-09 10:14:09.826071692 +0200
+++ b/atspi/atspi-gmain.c	2016-08-09 10:15:23.334436201 +0200
@@ -41,8 +41,8 @@
  */
 typedef struct
 {
-  GSource source; /**< the parent GSource */
-  DBusConnection *connection; /**< the connection to dispatch */
+  GSource source; /*< the parent GSource */
+  DBusConnection *connection; /*< the connection to dispatch */
 } DBusGMessageQueue;
 
 static gboolean message_queue_prepare  (GSource     *source,
@@ -95,11 +95,11 @@
 
 typedef struct
 {
-  GMainContext *context;      /**< the main context */
-  GSList *ios;                /**< all IOHandler */
-  GSList *timeouts;           /**< all TimeoutHandler */
-  DBusConnection *connection; /**< NULL if this is really for a server not a connection */
-  GSource *message_queue_source; /**< DBusGMessageQueue */
+  GMainContext *context;      /*< the main context */
+  GSList *ios;                /*< all IOHandler */
+  GSList *timeouts;           /*< all TimeoutHandler */
+  DBusConnection *connection; /*< NULL if this is really for a server not a connection */
+  GSource *message_queue_source; /*< DBusGMessageQueue */
 } ConnectionSetup;
 
 
@@ -510,7 +510,7 @@
   return cs;
 }
 
-/** @} */ /* End of GLib bindings internals */
+/* @} */ /* End of GLib bindings internals */
 
 /**
  * atspi_dbus_connection_setup_with_g_main: (skip)
